gui/src/mainwindow.cpp

```diff
@@ -0,0 +1,1459 @@
+            std::clog << "Pixel Size height: " << geoImage->size().height << std::endl;
+            std::clog << "Total size: " << (uint64_t)geoImage->size().width*geoImage->size().height << std::endl;
+
+            if((uint64_t)geoImage->size().width*geoImage->size().height > std::numeric_limits<int>::max()) {
+                std::clog << "Pixel size is too large" << std::endl;
+                hasValidBboxSize = false;
+            }
+            else {
+                hasValidBboxSize = true;
+            }
+        }
+        catch(dg::deepcore::Error e) {
+            std::string serverMessage(e.what());
+            std::clog << serverMessage << std::endl;
+            //check for invalid token message, first from DGCS, then from MapsAPI
+            if (serverMessage.find("INVALID CONNECT ID") != std::string::npos ||
+                serverMessage.find("Not Authorized - Invalid Token") != std::string::npos ||
+                serverMessage.find("Not Authorized - No Token") != std::string::npos) {
+                error += "Invalid web service token\n\n";
+                ui->tokenLineEdit->setStyleSheet("color: red;"
+                                                 "border: 1px solid red;");
+            }
+            //check for invalid username/password message
+            else if(serverMessage.find("This request requires HTTP authentication") != std::string::npos) {
+                error += "Invalid web service username and/or password\n\n";
+                ui->passwordLineEdit->setStyleSheet("color: red;"
+                                                    "border: 1px solid red;");
+                ui->usernameLineEdit->setStyleSheet("color: red;"
+                                                    "border: 1px solid red;");
+            }
+            //check for invalid map id
+            else if(serverMessage.find("Not Found") != std::string::npos) {
+                 error += "Invalid Map Id\n\n";
+                 ui->mapIdLineEdit->setStyleSheet("color: red;"
+                                                           "border: 1px solid red;");
+            }
+            else {
+                error += "Unknown web service authentication error occurred\n\n";
+            }
+            validJob = false;
+        }
+    }
+    statusBar()->clearMessage();
+
+    if(hasValidBboxSize == false){
+        if(ui->imageSourceComboBox->currentText() != "Local Image File") {
+            error += "The entered bounding box is too large\n\n";
+        }
+        else if(ui->imageSourceComboBox->currentText() == "Local Image File" && hasValidLocalImagePath) {
+            error += "The entered image is too large\n\n";
+        }
+        validJob = false;
+    }
+
+    return validJob;
+}
+
+void MainWindow::exportConfig(const QString &filepath)
+{
+    std::stringstream configContents;
+    std::ofstream configFile;
+    configFile.open(filepath.toStdString() + ".cfg");
```

Comment:
depends.  If we're using a standard File Open dialog, we can control that and it will end up correct.  If we are making them type it in, then that's subject to all kinds of craptasticness, and we either deal with it or fail gracefully.